packages/material-ui/src/utils/chainPropType.js

```diff
@@ -0,0 +1,18 @@
+function chainPropType(propType, check) {
```

Comment:
I would vote for the reverse argument order since we're calling check first. 

What do you think about using `...validators` and then just iterating over them until we find the first error?

Reply:
I agree with reversing the order if it works with documentation generation.